src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -2611,7 +2611,36 @@ public ObjectId getHeadRev(String url, String branchSpec) throws GitException, I
         return references;
     }
 
-    //
+    @Override
+    public Map<String, String> getRemoteSymbolicReferences(String url, String pattern)
+            throws GitException, InterruptedException {
+        ArgumentListBuilder args = new ArgumentListBuilder("ls-remote");
+        args.add("--symref");
+        args.add(url);
+        if (pattern != null) {
+            args.add(pattern);
+        }
+
+        StandardCredentials cred = credentials.get(url);
+        if (cred == null) cred = defaultCredentials;
+
+        String result = launchCommandWithCredentials(args, null, cred, url);
+
+        Map<String, String> references = new HashMap<>();
+        String[] lines = result.split("\n");
+        Pattern symRefPattern = Pattern.compile("^ref:\\s+([^ ]+)\\s+([^ ]+)$");
+        for (String line : lines) {
+            Matcher matcher = symRefPattern.matcher(line);
+            if (matcher.matches()) {
+                references.put(matcher.group(2), matcher.group(1));
+            } else  if (line.length() < 41) {
```

Comment:
If it helps, you can refer to my https://github.com/MarkEWaite/git-client-plugin/tree/master-PR236-resolve-symbolic-refs branch for some additional tests I added and for the `line.isEmpty()` change.